modules/javafx.graphics/src/main/java/javafx/scene/robot/Robot.java

```diff
@@ -0,0 +1,373 @@
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package javafx.scene.robot;
+
+import static com.sun.javafx.FXPermissions.CREATE_ROBOT_PERMISSION;
+
+import java.util.Objects;
+
+import javafx.geometry.Point2D;
+import javafx.geometry.Rectangle2D;
+import javafx.scene.image.Image;
+import javafx.scene.image.WritableImage;
+import javafx.scene.input.KeyCode;
+import javafx.scene.input.MouseButton;
+import javafx.scene.paint.Color;
+
+import com.sun.glass.ui.Application;
+import com.sun.glass.ui.GlassRobot;
+import com.sun.javafx.tk.Toolkit;
+
+/**
+ * A {@code Robot} is used for simulating user interaction such as
+ * typing keys on the keyboard and using the mouse as well as capturing
+ * graphical information without requiring a {@link javafx.scene.Scene}
+ * instance. Robot objects must be constructed and used on the JavaFX
+ * Application Thread.
+ *
+ * @since 11
+ */
+public final class Robot {
+
+    private final GlassRobot peer;
+
+    /**
+     * Constructs a new {@code Robot} that can be used for simulating user
+     * interactions. If a security manager is present, the application must
+     * have the {@link javafx.util.FXPermission} {@code "createRobot"} permission
+     * in order to construct a {@code Robot} object.
+     *
+     * @throws IllegalStateException if this object is constructed on a thread
+     * other than the JavaFX Application Thread.
+     * @throws SecurityException if a security manager exists and the application
+     * does not have the {@link javafx.util.FXPermission} {@code "createRobot"}
+     * permission.
+     */
+    public Robot() {
+        Application.checkEventThread();
+
+        // Ensure we have proper permission for creating a robot.
+        final SecurityManager sm = System.getSecurityManager();
+        if (sm != null) {
+            sm.checkPermission(CREATE_ROBOT_PERMISSION);
+        }
+
+        peer = Toolkit.getToolkit().createRobot();
+        Objects.requireNonNull(peer);
+        peer.create();
+    }
+
+    /**
+     * Presses the specified {@link KeyCode} key.
+     *
+     * @param keyCode the key to press
+     * @throws IllegalStateException if this method is called on a thread
+     * other than the JavaFX Application Thread.
+     */
+    public void keyPress(KeyCode keyCode) {
```

Comment:
npe if KeyCode is null?

Reply:
Shall I go ahead and make this change, Kevin?

Reply:
Throwing NPE if keyCode is null seems good to me.